adhocteam/Head-Start-TTADP
# Pull grant and grantee data from HSES

**Description of change**
Adding a cron job, which is part of the server and runs everyday at midnights.
The job downloads data from HSES (in production env), extracts the zip archive into "temp" directory. Parses the xml data and populates or updates the SmartHub db with downloaded data.

**How to test**
Make sure the tests pass.
The imported data can be seen on sandbox in the database using the following:
- open SSH tunnel:  `cf connect-to-service -no-client tta-smarthub-sandbox ttahub-sandbox`
- use your favorite postgres client with the parameters listed (if can't connect with `localhost`, use `127.0.0.1`)
- verify that there are grantees and grants in tables

(bonus test) additionally, to verify the download:
-  modify one of the HSES data download environment variables:
`HSES_DATA_FILE_URL, HSES_DATA_USERNAME, HSES_DATA_PASSWORD`
- modify the cron's schedule to '*/5 * * * *' in the `app.js` (run every 5 minutes instead of at midnight)
- deploy a branch to sandbox
- observe that there will be an error message

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/88

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
